lib/tail.js

```diff
@@ -3,64 +3,116 @@
 'use strict';
 
 const events = require('events');
-const childProcess = require('child_process');
-const tailStream = require('fs-tail-stream');
+const fs = require('fs');
+const TailLib = require('tail').Tail;
+const readLastLines = require('read-last-lines');
 const util = require('util');
 const CBuffer = require('CBuffer');
 const byline = require('byline');
-const commandExistsSync = require('command-exists').sync;
+
+const NB_OF_LINE_TO_PREFETCH = 50;
+const TAIL_RETRY_DELAY = 2000;
+const NEW_LINE_REGEX = /[\r]{0,1}\n/;
+const IS_WIN = process.platform === 'win32';
+
+function getPaddedLinePrefix(filePath, filePaths) {
+  const paddingLength = Math.max(...filePaths.map((fileName) => fileName.length));
+  return filePath.padStart(paddingLength, ' ');
+}
+
+const fileLogger = (fileName) => ({
+  info: (...data) => {
+    console.info(fileName, ':', ...data);
+  },
+  error: (...data) => {
+    console.error(fileName, ':', ...data);
+  }
+});
+
+function readLastLinesIfPossible(path, onLineCb) {
```

Comment:
I had to add this function, because the native tail does it. But the JS lib does not